www/class/config-generate/hostcategory.class.php

```diff
@@ -0,0 +1,132 @@
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of Centreon choice, provided that
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+class HostCategory extends AbstractObject
+{
+    /** @var array<int,mixed> */
+    private $hostCategories = [];
+
+    /** @var string */
+    protected $generate_filename = 'tags.cfg';
+    /** @var string */
+    protected $object_name = 'tag';
+    /** @var string */
+    protected $attributesSelect = '
+        hc_id as id,
+        hc_name as name
+    ';
+    /** @var string[] */
+    protected $attributes_write = [
+        'id',
+        'name',
+        'type',
+    ];
+
+    /**
+     * @param int $hostCategoryId
+     * @return self
+     */
+    private function addHostCategoryToList(int $hostCategoryId): self
+    {
+        $stmt = $this->backend_instance->db->prepare(
+            "SELECT {$this->attributesSelect}
+            FROM hostcategories
+            WHERE hc_id = :hc_id
+            AND level IS NULL
+            AND hc_activate = '1'"
+        );
+        $stmt->bindParam(':hc_id', $hostCategoryId, PDO::PARAM_INT);
+        $stmt->execute();
+        if ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
+            $this->hostCategories[$hostCategoryId] = $row;
+            $this->hostCategories[$hostCategoryId]['members'] = [];
+        }
+
+        return $this;
+    }
+
+    /**
+     * Add a host to members of a host category
+     *
+     * @param int $hostCategoryId
+     * @param int $hostId
+     * @param string $hostName
+     * @return self
+     */
+    public function addHostToHostCategoryMembers(int $hostCategoryId, int $hostId, string $hostName): self
+    {
+        if (! isset($this->hostCategories[$hostCategoryId])) {
+            $this->addHostCategoryToList($hostCategoryId);
+        }
+        if (
+            isset($this->hostCategories[$hostCategoryId])
+            && ! isset($this->hostCategories[$hostCategoryId]['members'][$hostId])
+        ) {
+            $this->hostCategories[$hostCategoryId]['members'][$hostId] = $hostName;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Write hostcategories in configuration file
+     */
+    public function generateObjects(): void
+    {
+        $this->seekFileEnd();
+        foreach ($this->hostCategories as $id => &$value) {
+            if (! isset($value['members']) || count($value['members']) === 0) {
+                continue;
+            }
+            $value['type'] = 'hostcategory';
```

Comment:
This is a ''magic value'' (hostcategory). Maybe you could create a meaningful class constant ?